src/core/hadoop/src/main/scala/HadoopUtils.scala

```diff
@@ -98,9 +98,10 @@ class SamplePathFilter extends Configured with PathFilter {
 
   override def accept(path: Path): Boolean = {
     // Note: checking fileSystem.isDirectory is very slow here, so we use basic rules instead
+    val double = random.nextDouble()
     !SamplePathFilter.isFile(path) ||
       (SamplePathFilter.isZipFile(path) && inspectZip) ||
-      random.nextDouble() < sampleRatio
+      double < sampleRatio
```

Comment:
double looks like a type name a la ``Double``. Rename to ``doubleValue`` or ``randomValue`` pershaps?